src/main/java/org/apache/maven/shared/dependency/analyzer/DependencyAnalyzer.java

```diff
@@ -39,4 +39,7 @@
 
     Set<String> analyze( URL url )
         throws IOException;
+
+    Set<DependencyUsage> analyzeWithUsages( URL url )
```

Comment:
This is an incompatible change. Would it make sense to simply make analyze include usages?

Reply:
> Would it make sense to simply make analyze include usages?

How were you thinking that would work, change the return type of `analyze` method? Isn't that also incompatible? And just to confirm, you're thinking about the case where someone wrote a custom implementation of `DependencyAnalyzer` and registered it for use with the dependency plugin?

One option that comes to mind is making `analyzeWithUsages` a default interface method with an implementation that throws something like `UnsupportedOperationException`. That way the dependency plugin can try to gracefully handle this case and fall back to the regular `analyze` method, for example:
```java
  try 
  {
    return dependencyAnalyzer.analyzeWithUsages( url );
  }
  catch ( UnsupportedOperationException e )
  {
    log.warn( "Dependency analyzer {} should be updated to implement analyzeWithUsages", dependencyAnalyzer  );
    return dependencyAnalyzer.analyze( url );
  }
```

Reply:
We don't have default interface methods until java 8. :-(

Reply:
In that case another option is to keep things as-is and have the dependency plugin use the same pseudo-code as above, except it would catch something like `NoSuchMethodError` instead of `UnsupportedOperationException` (with a comment explaining the situation). Certainly not ideal, but it seems like we ultimately need some way for the dependency plugin to detect that the analyzer doesn't the new functionality and gracefully degrade